lib/IRGen/IRGenModule.cpp

```diff
@@ -980,7 +983,7 @@ void IRGenModule::emitAutolinkInfo() {
                                  llvm::GlobalValue::CommonLinkage,
                                  llvm::Constant::getNullValue(Int1Ty),
                                  buf.str());
-    if (Triple.isOSBinFormatCOFF())
+    if (Triple.isOSBinFormatCOFF() && !Triple.isOSCygMing())
```

Comment:
This is repeated so much I think we should just have a predicate for it on IRGenModule.